// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Configuration for the various gammatone filterbanks.

syntax = "proto3";

package eidos.audition;

enum GammatoneFilterbankType {
  // Malcolm Slaney's filterbank. See
  //   - M. Slaney (Apple TR #35), "An Efficient Implementation of the
  //     Patterson-Holdsworth Cochlear Filter Bank.", 33-34.
  GAMMATONE_FILTERBANK_SLANEY = 0;

  // This gammatone filter is based on the implementation by Ning Ma from
  // University of Sheffield who, in turn, based his implementation on an
  // original algorithm from Martin Cooke's Ph.D thesis (Cooke, 1993) using
  // the base-band impulse invariant transformation. See
  //   - http://www.dcs.shef.ac.uk/~ning/resources/gammatone
  GAMMATONE_FILTERBANK_COOKE_AND_MA = 1;
};

// Next available ID: 2
message GammatoneFilterbankConfig {
  // Type of the filterbank.
  GammatoneFilterbankType filter_type = 1;
}

// Local Variables:
// mode: c++
// End:
